src/main/java/de/tum/in/www1/artemis/service/user/UserService.java

```diff
@@ -474,6 +493,25 @@ protected void deleteUser(User user) {
         learningGoalProgressRepository.deleteAllByUserId(user.getId());
         exerciseHintActivationRepository.deleteAllByUser(user);
 
+        // deleting Posts belonging to the user and belonging to the conversations created by the user
+        var conversationsByUser = conversationRepository.findAllByCreator(user);
+        conversationsByUser.forEach(conversation -> postRepository.deleteAllByConversationId(conversation.getId()));
+        postRepository.deleteAllByAuthor(user);
+
+        // deleting Posts belonging to all OneToOneChats where the user is a participant
+        var conversationParticipantEntities = conversationParticipantRepository.findConversationParticipantsByUser(user);
+        var oneToOneChatsToDelete = new HashSet<OneToOneChat>();
+        conversationParticipantEntities.forEach(participant -> oneToOneChatsToDelete.addAll(oneToOneChatRepository.findAllByConversationParticipantsContaining(participant)));
+        oneToOneChatsToDelete.forEach(oneToOneChat -> postRepository.deleteAllByConversationId(oneToOneChat.getId()));
+
+        // deleting Conversations and ConversationParticipants created by the user
+        conversationRepository.deleteAllByCreator(user);
+        conversationParticipantRepository.deleteAllByUser(user);
+
+        // deleting OneToOneChats where the user is a participant
+        oneToOneChatRepository.deleteAll(oneToOneChatsToDelete);
```

Comment:
I’d move this up to the `// deleting Posts belonging to all OneToOneChats where the user is a participant`-block. I think that makes it easier to follow since then all one-to-one-chat actions are resolved before moving on to the next part.